src/components/TodoList/TodoList.js

```diff
@@ -1,38 +1,67 @@
 import React from 'react';
 import './TodoList.css';
 
-function handleClick() {
-  console.log(this)
-}
-
 class TodoList extends React.Component {
 
-  // state = {
-  //   list: [
-  //     { id: 1, name: 'Fumer des cannes', active: true },
-  //     { id: 2, name: 'Coder cette shit', active: true },
-  //     { id: 3, name: 'Nager avec des baleines', active: true },
-  //     { id: 4, name: 'Rouler des cannes pour demain', active: true }
-  //   ],
-  //   newTaskName: "",
-  // // }
-  //
-  // handleClick = () => {
-  //   console.log(this)
-  // }
+  state = {
+    list: [],
+    newTaskName: "",
+    editTaskName: ""
+  }
 
   deleteTodo(itemToDeleteId) {
     const newList = this.state.list.filter((item) => item.id !== itemToDeleteId)
     this.setState({ list: newList })
   }
 
   createNewTask() {
+    if (this.state.list.length === 0) {
+      this.setState({
+        list: [{ id: 1, name: this.state.newTaskName, active: true, editing: false }],
+        newTaskName: ""
+      })
+      return;
+    }
+
     this.setState({
       list: [
         ...this.state.list,
-        { id: this.state.list[this.state.list.length - 1].id + 1, name: this.state.newTaskName, active: true }
+        { id: this.state.list[this.state.list.length - 1].id + 1, name: this.state.newTaskName, active: true, editing: false}
       ],
-      newTaskName: ""
+      newTaskName: "",
+      editTaskName: ""
```

Comment:
This seems not necessary, `editTaskName` does not need to be reset each time we create a task. This behaviour has little to do with task edition when you think about it